### PR TITLE
Google Calendar integration

### PR DESCRIPTION
**Pull request by MarkBrosche**

### CODE REVIEW

**Comment by zfox23** [comment]
@MarkBrosche, did you mean for this PR to stay up for now given that we are changing it a bunch?

File: usefulUtilities/hiFiCalendar/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# High Fidelity Calendar
+This app lets you integrate with the Google Calendar API to query meeting room schedules.
+
+# Setup Instructions
+1. Setup `hiFiCalendar.html` by copying it to the top level of your C:/ drive and opening a powershell window. Type the following command and press enter, `http-server -p 80` (You must have Node.js installed).
+2. In [Google's console API](https://console.developers.google.com/projectselector2/apis/credentials?supportedpurview=project), setup credentials to get an API Key, Client ID, and Client Secret to paste in `hiFiCalendar.html` (WARNING: DO NOT SHARE THESE OR POST THEM IN A PUBLIC PLACE).  Define your permitted access locations to include http://localhost under Domain Verification.
+
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/README.md]:
```suggestion
\`\`\`
```

File: usefulUtilities/hiFiCalendar/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        // Client ID and API key from the Developer Console
+        var CLIENT_ID = '';
+        var CLIENT_SECRET = '';
+        var API_KEY = '';
+
+3. Copy the calendar IDs into the html file from your Google Calendar, they'll look something like this:
+
+        YourGoogleDomain123456789@resource.calendar.google.com
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/README.md]: wrap this in \`\`\`

File: usefulUtilities/hiFiCalendar/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+4. In Interface, create a "Text" entity with the same name you've given your calendar. Paste the server script URL into the `serverScript` field in Entity Properties.
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/README.md]: do you also need to add a child entity to that label text entity that'll display the calendar schedule?

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        "clearEventList"
+    ];
+
+
+    this.preload = function(entityID) {
+        that = this;
+        that.entityID = entityID;
+        that.entityProperties = Entities.getEntityProperties(that.entityID, ['id', 'name', 'type', 'parentID']);
+        that.room = {};
+        if (that.entityProperties.name.indexOf("_SCHEDULE") === -1 &&
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: do you still need the server script to go on the schedule text entity?

Comment (zfox23) [reply usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: also, you don't have to change this in this pr, but for your next PR (as we discussed): please make the different entities in this system have different server scripts on them

File: usefulUtilities/hiFiCalendar/resources/hiFiCalendar.html [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+      pre.appendChild(textContent);
+    }
+
+
+    // Taken from https://stackoverflow.com/a/11318669
+    // Example date format: 2019-04-08T13:00:00-07:00
+    var googleDate = /^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})([+-]\d{2}):(\d{2})$/;
+    var MINS_PER_HOUR = 60;
+
+    function googleDateToUTCDate(d) {
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/resources/hiFiCalendar.html]: add comment explaining what function does below line 115. also remove whitespace on line 118

File: usefulUtilities/hiFiCalendar/resources/hiFiCalendar.html [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+      return new Date(year, month - 1, day, hour, minute - tzOffset, second, msec);
+    };
+
+
+    /**
+     * Print the summary and start datetime/date of the next ten events in
+     * the authorized user's calendar. If no events are found an
+     * appropriate message is printed.
+     */
+
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/resources/hiFiCalendar.html]: remove whitespace

File: usefulUtilities/hiFiCalendar/resources/hiFiCalendar.html [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+    }, {
+      "id": jakkuId
+    }, {
+      "id": narniaId
+    }, {
+      "id": ozId
+    }, {
+      "id": pandoraId
+    }];
+
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/resources/hiFiCalendar.html]: remove whitespace

File: usefulUtilities/hiFiCalendar/resources/hiFiCalendar.html [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+            appendPre('No upcoming events found.');
+            EventBridge.emitWebEvent(JSON.stringify({
+              type: "SEND_SCHEDULE",
+              room: calendar.id
+            }));
+          }
+        });
+      });
+    }
+
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/resources/hiFiCalendar.html]: add one more line of whitespace and add function comment

File: usefulUtilities/hiFiCalendar/resources/hiFiCalendar.html [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+    function onScriptEventReceived(data) {
+      var data = JSON.parse(data);
+      switch (data.type) {
+        case "UPDATE_SCHEDULE":
+          listUpcomingEvents();
+          break;
+      }
+    }
+
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/resources/hiFiCalendar.html]: add one more line of whitespace and add function comment

File: usefulUtilities/hiFiCalendar/resources/hiFiCalendar.html [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+    var EVENT_BRIDGE_SETUP_DELAY = 100;
+
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/resources/hiFiCalendar.html]: remove whitespace

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js (hunk 4 above)
Comment (MarkBrosche) [reply usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: We could either go with an empty server script on the _SCHEDULE entity or get rid of the room label entity altogether and simply change the background color of the _SCHEDULE entity.  Or we could leave it as is for now and decide in the V2 iteration.

Comment (zfox23) [reply usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: the latter sounds fine with me - please make sure you make a note of that for your v2 development!

**Pull request merged**